src/hotspot/share/gc/shared/collectedHeap.hpp

```diff
@@ -207,6 +209,10 @@ class CollectedHeap : public CHeapObj<mtGC> {
   }
 
  public:
+  MemRegion reserved_region() const {
+    assert(!UseZGC, "Not supported for ZGC");
+    return _reserved;
+  }
```

Comment:
Instead of asserting, would it be better to return an empty null range, and document such that "if null range is returned, this method is unsupported"? That way we don't need ZGC specific conditionals in shared code.